### PR TITLE
[SAI PTF]Syncd-rpc and PTF docker support sai ptf v2

### PR DESCRIPTION
**Pull request by richardyu-ms**
#### Why I did it
Make the ptf-sai docker which can suppor the sai-ptf v2
Make syncd rpc docker which supports sai-ptf v2

#### How I did it
change the docker make file

#### How to verify it
https://github.com/sonic-net/sonic-buildimage/pull/12619

local bulild the target

NOSTRETCH=y NOJESSIE=y make configure PLATFORM=vs
NOSTRETCH=y NOJESSIE=y NOBULLSEYE=y SAITHRIFT_V2=y make target/docker-ptf-sai.gz

NOSTRETCH=y NOJESSIE=y make configure PLATFORM=vs
NOSTRETCH=y NOJESSIE=y NOBULLSEYE=y make target/docker-ptf.gz

NOSTRETCH=y NOJESSIE=y make configure PLATFORM=broadcom
NOSTRETCH=y NOJESSIE=y ENABLE_SYNCD_RPC=y SAITHRIFT_V2=y make target/docker-syncd-brcm-rpcv2.gz
NOSTRETCH=y NOJESSIE=y ENABLE_SYNCD_RPC=y SAITHRIFT_V2=y make target/docker-saiserverv2-brcm.gz

NOSTRETCH=y NOJESSIE=y make configure PLATFORM=broadcom
NOSTRETCH=y NOJESSIE=y ENABLE_SYNCD_RPC=y make target/docker-syncd-brcm-rpc.gz
NOSTRETCH=y NOJESSIE=y ENABLE_SYNCD_RPC=y make target/docker-saiserver-brcm.gz

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205

### CODE REVIEW

File: platform/broadcom/docker-syncd-brcm-rpc/Dockerfile.j2 [hunk 1]
```diff
@@ -25,6 +25,7 @@ RUN apt-get update \
     cmake               \
     libqt5core5a        \
     libqt5network5      \
+    gdb                 \
```
[the remark this answers is not in the document]
Comment (richardyu-ms) [reply platform/broadcom/docker-syncd-brcm-rpc/Dockerfile.j2:28]: added the debug deps and add that dbg image to docker set

File: platform/broadcom/docker-saiserver-brcm.mk [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 DOCKER_SAISERVER_BRCM = docker-saiserver$(SAITHRIFT_VER)-brcm.gz
 $(DOCKER_SAISERVER_BRCM)_PATH = $(PLATFORM_PATH)/docker-saiserver-brcm
-$(DOCKER_SAISERVER_BRCM)_DEPENDS += $(SAISERVER)
+$(DOCKER_SAISERVER_BRCM)_DEPENDS += $(SAISERVER) $(SAISERVER_DBG)
```
Comment (xumia) [inline platform/broadcom/docker-saiserver-brcm.mk:5]: Can we add it only when DBG flag enabled?

File: .azure-pipelines/azure-pipelines-build.yml [hunk 3]
```diff
@@ -120,7 +120,8 @@ jobs:
               make $BUILD_OPTIONS ENABLE_ASAN=y target/docker-sonic-vs.gz
               mv target/docker-sonic-vs.gz target/docker-sonic-vs-asan.gz
             fi
-            make $BUILD_OPTIONS target/docker-sonic-vs.gz target/sonic-vs.img.gz target/docker-ptf.gz
+            make $BUILD_OPTIONS target/docker-sonic-vs.gz target/sonic-vs.img.gz target/docker-ptf.gz target/docker-ptf-sai.gz
+            make $BUILD_OPTIONS ENABLE_SYNCD_RPC=y SAITHRIFT_V2=y target/docker-ptf.gz target/docker-ptf-sai.gz
```
Comment (xumia) [inline .azure-pipelines/azure-pipelines-build.yml:124]: Can we add only for PR build?

File: .azure-pipelines/azure-pipelines-build.yml [hunk 4]
```diff
@@ -120,7 +120,8 @@ jobs:
               make $BUILD_OPTIONS ENABLE_ASAN=y target/docker-sonic-vs.gz
               mv target/docker-sonic-vs.gz target/docker-sonic-vs-asan.gz
             fi
-            make $BUILD_OPTIONS target/docker-sonic-vs.gz target/sonic-vs.img.gz target/docker-ptf.gz
+            make $BUILD_OPTIONS target/docker-sonic-vs.gz target/sonic-vs.img.gz target/docker-ptf.gz target/docker-ptf-sai.gz
+            make $BUILD_OPTIONS ENABLE_SYNCD_RPC=y SAITHRIFT_V2=y target/docker-ptf-sai.gz
```
Comment (xumia) [inline .azure-pipelines/azure-pipelines-build.yml:124]: The buld will be skipped, since the target  target/docker-ptf-sai.gz has already build on 123 line

**Review by xumia — APPROVED** [review]

**Pull request merged**